sentry-android-core/src/main/java/io/sentry/android/core/performance/AppStartMetrics.java

```diff
@@ -195,7 +223,46 @@ public static void onApplicationCreate(final @NotNull Application application) {
     final @NotNull AppStartMetrics instance = getInstance();
     if (instance.applicationOnCreate.hasNotStarted()) {
       instance.applicationOnCreate.setStartedAt(now);
-      instance.appLaunchedInForeground = ContextUtils.isForegroundImportance();
+      instance.appLaunchedInForeground =
+          instance.appLaunchedInForeground || ContextUtils.isForegroundImportance();
+      instance.registerApplicationForegroundCheck(application);
+    }
+  }
+
+  /**
+   * Register a callback to check if an activity was started after the application was created
+   *
+   * @param application The application object to register the callback to
+   */
+  public void registerApplicationForegroundCheck(final @NotNull Application application) {
```

Comment:
nice ❤️ can we also call this method in SentryAndroid then, so it's also covered? We probably will have to also keep a boolean flag to check if the callbacks were already `isRegistered`, to prevent this class from being added multiple times to the lifecycle callbacks (it's just an [ArrayList](https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/android/app/Application.java;l=58) from what I can see, so may have duplicates).